private/runtime/binding-spec.rkt

```diff
@@ -299,7 +316,18 @@
 
     [(suspend pv)
      (for/pv-state-tree ([stx pv])
-       (make-suspension (add-scopes stx local-scopes) (current-def-ctx)))]))
+       (make-suspension (add-scopes stx local-scopes) (current-def-ctx)))]
+    [(ellipsis pvs spec)
+     ; filter and split the environments
+     (define sts (exp-state-split/ellipsis st pvs))
+     ; expand on each sub-environment
+     (define sts^
+       (for/list ([st sts])
+         (simple-expand-internal spec st local-scopes)))
+     ; unsplit the sub-environments and merge that into the initial env.
```

Comment:
There's something a bit subtle about the state merging that deserves a note. exp-st doesn't just contain values for the relevant pvars here---it also contains a nest state. So why is it okay that we're throwing that away? It's okay because `nested` is at depth 0, so the static check will ensure that it never occurs under an ellipsis and thus the nest-st will not change for any of these expansions.

I think merging the environments is an interesting enough operation to pull out into a helper with an explanatory comment.